frontend-react/src/components/MessageTracker/MessageReceivers.tsx

```diff
@@ -1,62 +1,479 @@
 import { ReceiverData } from "../../config/endpoints/messageTracker";
-import { formattedDateFromTimestamp } from "../../utils/DateTimeUtils";
-
-import { QualityFilters } from "./QualityFilters";
+import { parseFileLocation } from "../../utils/misc";
 
 type MessageReceiverProps = {
-    receiverDetails: ReceiverData[] | undefined;
+    receiverDetails: ReceiverData[];
+};
+
+interface NormalizedReceiverData {
+    Name: string;
+    Service: string;
+    Date: string;
+    ReportId: string;
+    Main: "Batch" | "Process" | "Ready";
+    Sub: string;
+    FileName: string;
+    TransportResults: string;
+}
+
+interface MessageReceiversRowProps {
+    receiver: NormalizedReceiverData;
+    activeColumn: string;
+    activeColumnSortOrder: string;
+    setModalText: ({ title, body }: { title: string; body: string }) => void;
+    modalRef: React.RefObject<ModalRef>;
+}
+
+interface MessageReceiversColProps {
+    columnHeaderTitle: string;
+    activeColumn: string;
+    setActiveColumn: (colTitle: string) => void;
+    activeColumnSortOrder: string;
+    setActiveColumnSortOrder: (sortOrder: string) => void;
+    filterIcon: JSX.Element;
+    rowSpan: number;
+}
+
+const dateTimeFormatter = new Intl.DateTimeFormat("en-US", {
+    month: "2-digit",
+    day: "2-digit",
+    year: "numeric",
+    hour: "2-digit",
+    minute: "2-digit",
+    second: "2-digit",
+});
+
+const ColumnDataEnum = {
+    Name: "Name",
+    Service: "Service",
+    Date: "Date",
+    ReportId: "Report Id",
+    Main: "Main",
+    Sub: "Sub",
+    FileName: "File Name",
+    TransportResults: "Transport Results",
+};
+
+const FilterOptionsEnum = {
+    None: "",
+    ASC: "asc",
+    DESC: "desc",
+};
+
+const StatusEnum = {
+    BATCH: "BATCH",
+    PROCESS: "PROCESS",
+    READY: "READY",
+};
+
+const MessageReceiversCol = ({
+    columnHeaderTitle,
+    activeColumn,
+    setActiveColumn,
+    activeColumnSortOrder,
+    setActiveColumnSortOrder,
+    filterIcon,
+    rowSpan,
+}: MessageReceiversColProps) => {
+    const isCurrentlyActiveColumn = columnHeaderTitle === activeColumn;
+    const handleColHeaderClick = () => {
+        if (!isCurrentlyActiveColumn) {
+            // Reset active column and sort order on new column click
+            setActiveColumn(columnHeaderTitle);
+            setActiveColumnSortOrder(FilterOptionsEnum.None);
+        }
+
+        // Explicitly set the proceeding sort order
+        switch (true) {
+            case activeColumnSortOrder === FilterOptionsEnum.None:
+                setActiveColumnSortOrder(FilterOptionsEnum.ASC);
+                break;
+            case activeColumnSortOrder === FilterOptionsEnum.ASC:
+                setActiveColumnSortOrder(FilterOptionsEnum.DESC);
+                break;
+            case activeColumnSortOrder === FilterOptionsEnum.DESC:
+                setActiveColumnSortOrder(FilterOptionsEnum.None);
+                break;
+        }
+    };
+    return (
+        <th
+            className={
```

Comment:
Can use `classnames` for this:

```tsx
className={ classnames("active-col-header: isCurrentlyActiveColumn && activeColumnSortOrder !== FilterOptionsEnum.None) }
```

Reply:
Sounds good, can refactor my dynamic class implementation across the board.